src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -468,6 +470,25 @@ public IndexCoordinates getIndexCoordinatesFor(Class<?> clazz) {
 		return getRequiredPersistentEntity(clazz).getIndexCoordinates();
 	}
 
+	@Override
+	public <T> UpdateResponse update(T entity) {
+		return update(this.buildUpdateQueryByEntity(entity), this.getIndexCoordinatesFor(entity.getClass()));
+	}
+
+	protected <T> UpdateQuery buildUpdateQueryByEntity(T entity) {
+		String id = this.getEntityId(entity);
+		Assert.notNull(entity, "entity must have an id that is notnull");
```

Comment:
```suggestion
		Assert.notNull(entity, "entity must not be null");
		
		String id = this.getEntityId(entity);
		Assert.notNull(id, "entity must have an id that is not null");
```

assert the entity and then the id